packages/design-system/src/components/web-components/ds-choice/ds-choice.test.tsx

```diff
@@ -0,0 +1,274 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import './ds-choice';
+
+const defaultAttrs = {
+  name: 'foo',
+  value: 'boo',
+  type: 'checkbox',
+  label: 'George Washington',
+  id: 'static-id',
+};
+
+function renderChoice(customAttrs = {}) {
+  return render(<ds-choice {...defaultAttrs} {...customAttrs} />);
+}
+
+describe('Choice', () => {
+  it('is a radio', () => {
+    const { asFragment } = renderChoice({ type: 'radio' });
+    expect(asFragment()).toMatchSnapshot();
+
+    const radioEl = screen.getByRole('radio');
+    expect(radioEl).toHaveAttribute('type', 'radio');
+  });
+
+  it('is a checkbox', () => {
+    const { asFragment } = renderChoice();
+    expect(asFragment()).toMatchSnapshot();
+
+    const checkboxEl = screen.getByRole('checkbox');
+    expect(checkboxEl).toHaveAttribute('type', 'checkbox');
+  });
+
+  it('is not checked', () => {
+    renderChoice();
+    const el = screen.getByRole('checkbox');
+    expect(el).not.toBeChecked();
+  });
+
+  it('is checked', () => {
+    renderChoice();
+    const el = screen.getByRole('checkbox');
+    userEvent.click(el);
+    expect(el).toBeChecked();
+  });
+
+  it('is checked (with prop)', () => {
+    renderChoice({ checked: true });
+    const el = screen.getByRole('checkbox');
+    expect(el).toBeChecked();
+  });
+
+  it('is defaultChecked', () => {
+    renderChoice({ 'default-checked': true });
+    const el = screen.getByRole('checkbox');
+    expect(el).toBeChecked();
+  });
+
+  it.skip('is required', () => {
```

Comment:
i'm noticing some strangeness when it comes to spreading props... they aren't being correctly applied to the underlying elements like we see on react components. i'm skipping these types of tests for now.